mountpoint-s3-crt/src/s3/client.rs

```diff
@@ -369,6 +367,14 @@ impl MetaRequestOptions {
         options.inner.part_size = part_size;
         self
     }
+
+    /// Set send_using_async_writes TODO: explain
+    pub fn send_using_async_writes(&mut self, send_using_async_writes: bool) -> &mut Self {
+        // SAFETY: we aren't moving out of the struct.
+        let options = unsafe { Pin::get_unchecked_mut(Pin::as_mut(&mut self.0)) };
+        options.inner.send_using_async_writes = send_using_async_writes;
```

Comment:
There is some detail here: https://github.com/awslabs/aws-c-s3/pull/418

Simplifying a bit: with the "pull-based" API, the CRT would start preparing a request (and occupy a slot) and then stall until it received data from Mountpoint. With enough PutObject meta-requests to fill all available "slots", it would indefinitely delay new requests and "deadlock" Mountpoint.

The new push-based API allows the CRT to only start preparing a request after it has received enough data from Mountpoint.